File: server/mbservice/src/main/java/org/rliz/mbs/artist/model/ArtistCredit.java
package org.rliz.mbs.artist.model;

import org.rliz.mbs.common.model.AbstractEntity;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;
import java.util.Set;

/**
 * Represents artist credit.
 */
@Entity
@Table(name = "artist_credit", indexes = {
        @Index(name = "ix_artistcredit_id", columnList = "id")
})
public class ArtistCredit extends AbstractEntity implements Serializable {

    @Column(name = "name")
    private String name;

    @Column(name = "artist_count")
    private Integer artistCount;

    @Column(name = "ref_count")
    private Integer refCount;

    @Column(name = "created")
    private Date created;

    @OneToMany(mappedBy = "artistCredit", fetch = FetchType.LAZY)
    private Set<ArtistCreditName> artistCreditName;

    public String getName() {
        return name;
    }

    public Integer getArtistCount() {
        return artistCount;
    }

    public Integer getRefCount() {
        return refCount;
    }

    public Date getCreated() {
        return created;
    }

    public Set<ArtistCreditName> getArtistCreditName() {
        return artistCreditName;
    }

}

File: server/mbservice/src/main/java/org/rliz/mbs/artist/model/ArtistCredit.kt
package org.rliz.mbs.artist.model

import org.rliz.mbs.common.model.AbstractEntity
import java.io.Serializable
import java.util.Date
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.Index
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(
    name = "artist_credit",
    indexes = [Index(name = "artist_credit_pkey", columnList = "id")]
)
class ArtistCredit : AbstractEntity(), Serializable {

    @Column(name = "name")
    val name: String? = null

    @Column(name = "artist_count")
    val artistCount: Int? = null

    @Column(name = "ref_count")
    val refCount: Int? = null

    @Column(name = "created")
    val created: Date? = null

    @OneToMany(mappedBy = "artistCredit", fetch = FetchType.LAZY)
    val artistCreditName: Set<ArtistCreditName>? = null
}
